### ISSUE TITLE
Filter with or

### ISSUE THREAD

**Issue by gabeno** [elastic/elasticsearch-dsl-py#493]
I have the following query that does what I need - returns docs matching `name` and satisfying filtering conditions where `country=US` or `country=NULL`.

```bash
{
    "query": {
        "bool": {
            "must": {
                "match": {
                    "name": "xxx"
                }
            },
            "filter": {
                "bool": {
                    "filter": [
                        {
                            "or": [
                                {"missing": {"field": "country"}},
                                {"match": {"country": "US"}}
                            ]
                        }
                    ]
                }
            }
        }
    }
}
```

I still have no luck translating this using __elasticsearch_dsl__, any ideas?

**Comment by HonzaKral** [comment]
I believe this is what you are looking for (since `country` is an exact match I switched it to a `term` filter):
```
s = Search()
s = s.query('match', name='xxx')
s = s.filter(Q('missing', field='country') | Q('term', country='US'))
```

**Comment by gabeno** [comment]
I tried this earlier. It translates to a `should` filter clause which gives other values for `country` besides the only desired values: `US` and `null`.

**Comment by HonzaKral** [comment]
And are you sure you have the correct mapping? When I do (in 5.0):
```
class MyDoc(DocType):
    name = Text()
    country = Keyword()
    class Meta:
        index = 'my-index'
MyDoc.init()
MyDoc(name='xxx yyy zzz', country='not US').save()
MyDoc(name='xxx yyy zzz', country='US').save()
MyDoc(name='xxx yyy zzz').save()
```
I can then run the search and get the correct response (just 2nd and 3rd document):
```
s = MyDoc.search()
s = s.query('match', name='xxx')
s = s.filter(~Q('exists', field='country') | Q('term', country='US'))
```

**Comment by HonzaKral** [comment]
It could also be different version, could you please try with `master`? Thanks!

**Comment by gabeno** [comment]
I am running version elasticsearch-dsl=__2.1.0__ on elasticsearch version __2.3.3__

Still no luck, but I get the gist of how to build the query. I will look at it more closely. Maybe there is something I am not doing right.

Further, I am curious if there is any benefits of using either `Q('missing', field='country')` or `~Q('exists', field='country')`. I see you interchanged them.

**Comment by HonzaKral** [comment]
that's just an artefact of me running on `5.0` where `missing` query has been removed in favor of `not exists`.

This is fixed in master (via 4b4487d0659ac8f2cf0e6499009ce696fca35684), and should be released with new version of elasticsearch-dsl.
